src/lib/assetBridger/erc20Bridger.ts

```diff
@@ -207,29 +218,75 @@ export class Erc20Bridger extends AssetBridger<
   }
 
   /**
-   * Approve tokens for deposit to the bridge. The tokens will be approved for the relevant gateway.
+   * Get a tx request to approve tokens for deposit to the bridge.
+   * The tokens will be approved for the relevant gateway.
    * @param params
    * @returns
    */
-  public async approveToken(
+  public async getApproveTokenRequest(
```

Comment:
getting the request currently requires a `l1Signer: Signer`
it would be cool to be able to instead provide a provider and from address.
The alternative would be to expect the consumer to use a [void signer](https://docs.ethers.io/v5/api/signer/#VoidSigner).

I think currently in deposits flow we instead surface the option to set a `from` address

Reply:
We dont actually need a from here since the actual approval will use msg.sender - so just replacing the l1signer with an l1provider and leaving at that